src/main/resources/public/jsx/components/SurveyCollection.js

```diff
@@ -72,14 +72,16 @@ export class SurveyCollection extends React.Component {
         const rules = this.props.surveyRules.filter(rule => (rule.questionId && rule.questionId === id)
             || (rule.questions && rule.questions.includes(id))
             || (rule.questionSetIds1 && (rule.questionSetIds1.includes(id) || rule.questionSetIds2.includes(id)))
-            || (rule.question1 && rule.question1 === id ||rule.question2 === id));
-        return rules.map((r) => {
-            return r.questionId ? r.regex? `${'Rule: ' + r.ruleType + ' | Question: ' + r.questionsText + ' | Regex: '+r.regex} ` : `${'Rule: '+r.ruleType + ' | Question: ' + r.questionsText + ' | Min: ' + r.min + ' | Max: ' + r.max} `
-                : r.questions ? `${'Rule: ' + r.ruleType + ' | Questions: ' + r.questionsText + ' | Valid Sum: ' + r.validSum} `
-                    :r.questionSetIds1 ? `${'Rule: ' + r.ruleType + ' | QuestionSet1: ' + r.questionSetText1 + ' | QuestionsSet2: ' + r.questionSetText2} `
-                        : `${'Rule: ' + r.ruleType + ' | Question1: ' + r.questionText1 + ' | Answer1: ' + r.answerText1 + ' | Question2: ' + r.questionText2 + ' | Answer2: ' + r.answerText2} `;
-
-        }).join('\n');
+            || (rule.question1 && rule.question1 === id || rule.question2 === id));
+        return rules.map((r) =>
+             r.questionId
```

Comment:
This is still not quite right.  Each new ? block will be indented.  Each ? should have the same indention level as its matching :. The matching pairs can be multiple lines apart (which is why this make it easier to read the logic).